test(compiler-cli): migrate template compliance tests

crisbeto opened the pull request:
Migrates the `r3_view_compiler_template` tests to the new format. Also introduces a new matcher for unique function names.

JoostK commented on packages/compiler-cli/test/compliance/test_cases/r3_view_compiler_template/unique_listener_function_names.js:
```diff
@@ -0,0 +1 @@
+…
```
This file can probably be removed as it's not listed as expectation in the test manifest (only the extra check verification). There's a few more empty files like this.

crisbeto replied:
The assertions all happen through the unique function checker, but IIRC I had to add these empty files, because the testing infrastructure was expecting a file to be there.

petebacondarwin replied:
I think that is correct. It is this file that triggers the check. Perhaps we should actually add something to these files to improve the test coverage?

crisbeto replied:
I've added a comment in the files about where the actual assertions happen.

JoostK reviewed (approved):
Nice! This needs a rebase though as the new extra check function is conflicting with a new one that landed on master.

crisbeto committed:
test(compiler-cli): migrate template compliance tests

Migrates the `r3_view_compiler_template` tests to the new format.
Also introduces a new matcher for unique function names.
